packages/providers/src.ts/formatter.ts

```diff
@@ -392,37 +388,74 @@ export class Formatter {
         return Formatter.check(this.formats.receiptLog, value);
     }
 
+    //fill the txReceipt obj
     receipt(value: any): TransactionReceipt {
         const result: TransactionReceipt = Formatter.check(this.formats.receipt, value);
 
-        // RSK incorrectly implemented EIP-658, so we munge things a bit here for it
-        if (result.root != null) {
-            if (result.root.length <= 4) {
-                // Could be 0x00, 0x0, 0x01 or 0x1
-                const value = BigNumber.from(result.root).toNumber();
-                if (value === 0 || value === 1) {
-                    // Make sure if both are specified, they match
-                    if (result.status != null && (result.status !== value)) {
-                        logger.throwArgumentError("alt-root-status/status mismatch", "value", { root: result.root, status: result.status });
-                    }
-                    result.status = value;
-                    delete result.root;
-                } else {
-                    logger.throwArgumentError("invalid alt-root-status", "value.root", result.root);
-                }
-            } else if (result.root.length !== 66) {
-                // Must be a valid bytes32
-                logger.throwArgumentError("invalid root hash", "value.root", result.root);
-            }
-        }
-
         if (result.status != null) {
             result.byzantium = true;
         }
 
         return result;
     }
 
+    txRecordToTxResponse(txRecord: HederaTransactionResponse): TransactionResponse {
+        return {
+            accessList: null,
+            chainId: 1, // get from provider for "mainnet", "testnet", "previewnet"?
```

Comment:
Yes, we must be able to place the `chainId` . You can map it outside of this function, or place it in the `record` argument and map it inside

Reply:
Placed in the `txRecord` & mapped here